### ISSUE TITLE
How to Add HTTP Endpoint to App Manifest?

### ISSUE THREAD

**Issue by jcburley** [MicrosoftDocs/msteams-docs#7410]
> Use the https endpoint provided by ngrok in your app manifest.

So...how does one do this? There's no link in that text, nor can I find "http" or "endpoint" in https://learn.microsoft.com/en-us/microsoftteams/platform/resources/schema/manifest-schema in any text that explains how to do this.

---
#### Document Details

⚠ *Do not edit this section. It is required for learn.microsoft.com ➟ GitHub issue linking.*

* ID: 45ca0926-65fc-66a5-b881-5b99a589f2ec
* Version Independent ID: b9e40da9-1029-f65f-18b5-f450a8ea87d2
* Content: [Test and debug your bot locally - Teams](https://learn.microsoft.com/en-us/microsoftteams/platform/bots/how-to/debug/locally-with-an-ide)
* Content Source: [msteams-platform/bots/how-to/debug/locally-with-an-ide.md](https://github.com/MicrosoftDocs/msteams-docs/blob/main/msteams-platform/bots/how-to/debug/locally-with-an-ide.md)
* Service: **msteams**
* GitHub Login: @surbhigupta
* Microsoft Alias: **anclear**

**Comment by sayali-MSFT** [comment]
@jcburley -Here is manifest sample available where ngrok/subdomain URI is updated.
https://learn.microsoft.com/en-us/microsoftteams/platform/resources/schema/manifest-schema#create-a-manifest-file 

If required, We will update the document accordingly.

**Comment by jcburley** [comment]
> @jcburley -Here is manifest sample available where ngrok/subdomain URI is updated. https://learn.microsoft.com/en-us/microsoftteams/platform/resources/schema/manifest-schema#create-a-manifest-file

Thanks, but which specific field has been updated? I see `webApplicationInfo.resource`, which seems closest to what I'm asking about – but, as we aren't using Azure for this particular project, there'll be no "Azure AD application ID", which is shown as "Required".

> If required, We will update the document accordingly.

Seems like a good idea to me.

Is there a video tutorial available showing how to build a simple Bot without using Azure? I'm trying to get this done, using ngrok and such, but am getting this error in VSCode, despite having created a brand-new Bot and using its id (for both the top-level and the bot Id fields), in which case there should be no conflict:

```
code:AppStudioPlugin.TeamsAppCreateConflict, message: Failed to create Teams app with 409 Conflict error. That may come from your app id is conflicting with another app in your tenant. Click Get Help to learn more.
```

I've tried the "Get Help" link, but am not finding it sufficiently helpful to resolve this issue.

(Our existing app, which will serve as the endpoint for authenticating a user who has just installed our forthcoming MS Teams Bot, is not running in Azure. We have other components that do, but they're not appropriate for us to use for our MS Teams integration.)

**Comment by ChetanSharma-msft** [comment]
Hello @jcburley - As an alternative to "Azure" for bot creation, you can use Bot Framework site (https://dev.botframework.com/bots) like below:
![image](https://user-images.githubusercontent.com/59052547/200495854-65dfb3a2-37ac-49fe-88f8-2c93d2bcc8d1.png)

Also, as per the error details:
Looks like you are trying to use same app id/client id for multiple apps.
Please try creating new and try it once.

**Comment by ChetanSharma-msft** [comment]
Hello @jcburley - Hope you are doing well!!
Please confirm whether your issue has resolved or need any further help here?

**Comment by jcburley** [comment]
> Hello @jcburley - Hope you are doing well!! Please confirm whether your issue has resolved or need any further help here?

Hi – I haven't tried your most-recent suggestions, as I had to put the MS Teams work on hold for awhile. 😞 

I'm also still waiting on Developer access (from our IT department), which has been a roadblock in a few different directions.

Note that there shouldn't be any app/client-id conflict, as I did try creating a brand-new one and using the resulting ID. Still got the same 409 error.

But it's possible the lack of a proper Azure Application ID was the problem.

I look forward to picking up this project again and making further progress.

Thanks for all the help!
